Pressure analysis

nlautrette opened the pull request:
added new file (pressure_analysis.py) containing function for detecting points in interest in pressure data

fazerlicourice7 committed:
setup data piping for Nolan

fazerlicourice7 committed:
force upload data file for Nolan

nlautrette committed:
Create pressure_analysis.py

Added file for signal detection function (currently test(dataamount, dataarray))

nlautrette committed:
new file for signal detection function

nlautrette committed:
Update pressure_analysis.py

nlautrette committed:
Update pressure_analysis.py

fazerlicourice7 commented on DataMonitoring/pressure_analysis.py:
```diff
@@ -0,0 +1,163 @@
+
+#function for detecting peaks in data
+##example: detect2(tankData)
+def detect2(data):
```
Can we rename this to `detectPeaks` Or is this actually detecting large changes? in which case we should call it `detectTransition` or something

fazerlicourice7 commented on DataMonitoring/pressure_analysis.py:
```diff
@@ -0,0 +1,163 @@
+    n = 5  # the larger n is, the smoother curve will be
+    b = [1.0 / n] * n
+    a = 1
+    yyy = lfilter(b,a,data)
```
what is this variable doing? Can we give it a descriptive name?

fazerlicourice7 commented on DataMonitoring/pressure_analysis.py:
```diff
@@ -0,0 +1,163 @@
+        peaksall.append(peaks2)
+    print("Total Peaks detected:", len(peaksall))
+    print(peaksall)
+    plt.figure()
+
+    plt.plot(dary)
+
+    plt.plot(dary_step/10)
+
```
How about this function returns the location of the transitions, and you have a separate function to plot it. This modularity will make it very easy to combine with our other code later.

fazerlicourice7 commented on DataMonitoring/pressure_analysis.py:
```diff
@@ -0,0 +1,163 @@
+    n = 20  # the larger n is, the smoother curve will be
+    b = [1.0 / n] * n
+    a = 1
+    yyyy = lfilter(b,a,dary_step2/10)
```
once again. naming

fazerlicourice7 commented on DataMonitoring/pressure_analysis.py:
```diff
@@ -0,0 +1,163 @@
+    print(peaks3)
+
+    # plots
+    plt.figure()
+
+    plt.plot(dary2)
+    #orange:
+    plt.plot(dary_step2/10)
```
see above. move plotting

fazerlicourice7 reviewed (changes requested):
Looks good! Just some minor changes.
Great job figuring out git without my help

nlautrette committed:
Create pressure analysis.ipynb

nlautrette committed:
Most recent version of pressure analysis code

Removed previous jupyter notebook entirely, added new one containing the code for reading in data and processing it, with separate functions for each different type of manipulation.  There is an example of the entire process at the bottom of the notebook

nlautrette committed:
Delete pressure_analysis.py
